Add recipe for atl-long-lines.

jcs090218 opened the pull request:
### Brief summary of what the package does

Automatically truncate lines for long lines.

### Direct link to the package repository

https://github.com/jcs-elpa/atl-long-lines

### Your association with the package

The maintainer.

### Relevant communications with the upstream package maintainer

None needed.

### Checklist

Please confirm with `x`:

- [x] The package is released under a [GPL-Compatible Free Software License](https://www.gnu.org/licenses/license-list.en.html#GPLCompatibleLicenses).
- [x] I've read [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [x] I've used the latest version of [package-lint](https://github.com/purcell/package-lint) to check for packaging issues, and addressed its feedback
- [x] My elisp byte-compiles cleanly
- [x] `M-x checkdoc` is happy with my docstrings
- [x] I've built and installed the package using the instructions in [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [ ] I have confirmed some of these without doing them

riscy commented:
Thanks, I like this idea.  About the summary line: "Automatically truncate lines for long lines" -- maybe we can make this clearer.  It seems to be turning truncate-lines off when the point is on a long line.

I recommend toggling truncate-lines-mode on or off on an idle timer, rather than synchronously in a post-command-hook -- and also to avoid redundant calls to e.g. `(toggle-truncate-lines 1)` when the user is on the same line, e.g. someone touch typing a sequence of characters.  Considerations like these might not seem important when you're writing a single package in isolation, but they add up, _especially_ when you're using post-command-hooks.

Going forward, I'd encourage you to not just copy-paste the summary line of your packages into your Commentary --  it's useful for users to see expanded descriptions, entrypoints, examples, etc.

Minor, but how come the function `atl-long-lines--turn-on-line-reminder-mode` has `-reminder-mode` in it?

jcs090218 commented:
> About the summary line: "Automatically truncate lines for long lines" -- maybe we can make this clearer. It seems to be turning truncate-lines off when the point is on a long line.

How about `Turn off truncate lines when the line is long`? I think is a bit clearer.

> I recommend toggling truncate-lines-mode on or off on an idle timer, rather than synchronously in a post-command-hook -- and also to avoid redundant calls to e.g. (toggle-truncate-lines 1) when the user is on the same line, e.g. someone touch typing a sequence of characters. Considerations like these might not seem important when you're writing a single package in isolation, but they add up, especially when you're using post-command-hooks.

Added! Thanks!

> Going forward, I'd encourage you to not just copy-paste the summary line of your packages into your Commentary -- it's useful for users to see expanded descriptions, entrypoints, examples, etc.

Yeah, that's my bad. I might improve it in the future version.

> Minor, but how come the function atl-long-lines--turn-on-line-reminder-mode has -reminder-mode in it?

Oops, it's a typo! Fixed!

riscy commented:
> How about Turn off truncate lines when the line is long? I think is a bit clearer.

Thanks for accommodating, that looks good to me.  I'd hyphenate truncate-lines (there's a variable called truncate-lines).

> Yeah, that's my bad. I might improve it in the future version.

Let's start doing this here, rather than making it a maintenance burden for you in the future.  You're an experienced package author and maintainer, and I feel like this is a corner you don't need to cut. 

Here's the latest checks --

byte-compile (using Emacs 27.1):
```
atl-long-lines.el:68:9:Warning: reference to free variable
    `atl-long-lines-mode'
```
This one's because the variable's referenced before the mode is declared -- easy fix.

checkdoc (using version 0.6.2):
- No issues!

package-lint (using version 20200906.512):
- No issues!

Other possible lints:
- atl-long-lines.el#L83: It's safer to sharp-quote function names; use `#'`
- atl-long-lines.el#L87: It's safer to sharp-quote function names; use `#'`

jcs090218 commented:
> Thanks for accommodating, that looks good to me. I'd hyphenate truncate-lines (there's a variable called truncate-lines).

Fixed! And thank you!

> Let's start doing this here, rather than making it a maintenance burden for you in the future. You're an experienced package author and maintainer, and I feel like this is a corner you don't need to cut.

Okay, I will think about starting from now! Thanks for the advice! 👍 

> Here's the latest checks --

All fixed!

riscy commented:
Looks good - feel free to ping when the Commentary is fleshed out.

jcs090218 commented:
Okay, I have added a commentary. I have checked other packages as references; and hopefully is good enough. Let me know if something else I may miss or improve!

And again! Thank you very much! ;)

riscy commented:
Thank you!  That Commentary looks perfect.  It's more effort to do this for each of your packages, but it pays off in the long run.
